PythonVisualizations/BinaryTreeBase.py

```diff
@@ -0,0 +1,515 @@
+
+   # unhighlight the right line of the node
+   def unHighlightRightLine(self, node):
+      # get all the items with the node's tag
+      nodeFamily = set(self.canvas.find_withtag(node.tag))
+      # get all the lines that are left
+      lines = set(self.canvas.find_withtag("right line"))
+
+      line = (lines & nodeFamily).pop()
+
+      self.canvas.itemconfig(line, fill="black")
+
+   # highlight the circle associated with the node
+   def highlightCircle(self, node, color="red"):
+      self.canvas.itemconfig(node.drawable.display_shape, outline=color)
+
+   # unhighlight the circle associated with the node
+   def unHighlightCircle(self, node):
+      self.canvas.itemconfig(node.drawable.display_shape, outline="black")
+
+   # creates a highlighted value on top of the normal value associated with the node
+   def createHighlightedValue(self, node, color="green"):
+      return self.canvas.create_text(node.coords[0], node.coords[1], text=node.getKey(), font=self.VALUE_FONT,
+                                               fill=color)
+
+   # highlight the value associated with the node
+   def highlightValue(self, node, color="red"):
+      self.canvas.itemconfig(node.drawable.display_val, fill=color)
+
+   # unhighlight the value associated with the node
+   def unHighlightValue(self, node):
+      self.canvas.itemconfig(node.drawable.display_val, fill="black")
+
+   # draws the circle and the key value
+   def createNodeShape(self, x, y, key, tag):
+      circle = self.canvas.create_circle(x, y, self.CIRCLE_SIZE, tag = tag)
+      circle_text = self.canvas.create_text(x,y, text=key, font=self.VALUE_FONT, tag = tag)
+
+      # get rid of the none shape that was in its spot
+      items = set(self.canvas.find_overlapping(*(self.canvas.coords(circle))))
+      noneShape = set(self.canvas.find_withtag("none"))
+      noneItems = (items & noneShape)
+      
+      while len(noneItems) > 0:
+         self.canvas.delete(noneItems.pop())
+
+      self.canvas.update()
+
+      return circle, circle_text
+
+   # creates the none shape that nodes will point to
+   # line coords should be the coords of the line that points to none
+   def createNoneShape(self, node, level, lineCoords, direction, tag):
+      y = lineCoords[3] + (self.CIRCLE_SIZE // 2)
+      xCoord = self.calculateCoordinates(node, level+1, direction)[0]
+
+      shrinkFactor = self.CIRCLE_SIZE // 3
+
+      coords1 = ( xCoord-self.CIRCLE_SIZE, y, 
+                  xCoord+self.CIRCLE_SIZE, y)
+      coords2 = (  shrinkFactor + coords1[0], coords1[1] + shrinkFactor,
+                  -shrinkFactor + coords1[2], coords1[3] + shrinkFactor)
+      coords3 = (  shrinkFactor + coords2[0], coords2[1] + shrinkFactor, 
+                  -shrinkFactor + coords2[2], coords2[3] + shrinkFactor)
+
+      self.canvas.create_line(coords1, tag = (tag, "none"))
+      self.canvas.create_line(coords2, tag = (tag, "none"))
+      self.canvas.create_line(coords3, tag = (tag, "none"))
```

Comment:
I think this method, `createNoneShape()`, won't be used based on my request to not show the empty child links.

Reply:
Implemented in commit ce09b89